sdks/python/apache_beam/yaml/yaml_join.py

```diff
@@ -0,0 +1,273 @@
+  error_prefix = f'Invalid value "{equalities}" for "equalities".'
+
+  valid_cols = {}
+  inputs = list(pcolls.keys())
+  for input in inputs:
+    fields = set()
+    for field in pcolls[input].element_type._fields:
+      fields.add(field[0])
+    valid_cols[input] = fields
+
+  if isinstance(equalities, str):
+    for input in valid_cols.keys():
+      if equalities not in valid_cols[input]:
+        raise ValueError(
+            f'{error_prefix} When "equalities" is a str, '
+            f'it must be a field name that exists in all the specified inputs.')
+    equality = {input: equalities for input in inputs}
+    return [equality]
+
+  if not isinstance(equalities, list):
+    raise ValueError(f'{error_prefix} It should be a str or a list.')
+
+  input_edge_list = []
+  for equality in equalities:
+    invalid_dict_error = ValueError(
+        f'{error_prefix} {equality} '
+        f'should be a dict[str, str] containing at least 2 items.')
+    if not isinstance(equality, dict):
+      raise invalid_dict_error
+    if len(equality) < 2:
+      raise invalid_dict_error
+
+    for input, col in equality.items():
+      if input not in inputs:
+        raise ValueError(
+            f'{error_prefix} "{input}" is not a specified alias in "input"')
+      if col not in valid_cols[input]:
+        raise ValueError(
+            f'{error_prefix} "{col}" is not a valid field in "{input}"')
+
+    input_edge_list.append(tuple(equality.keys()))
+
+  if not _is_connected(input_edge_list):
+    raise ValueError(
+        f'{error_prefix} '
+        f'The equalities provided does not ensure that '
+        f'all the inputs {inputs} are connected.')
+
+  return equalities
+
+
+def _parse_fields(tables, fields):
+  output_fields = []
+  named_columns = set()
+  for input, cols in fields.items():
+    error_prefix = f'Invalid input "{input}" for fields.'
+    if input not in tables:
+      raise ValueError(error_prefix)
+    if isinstance(cols, list):
+      for col in cols:
+        if col in named_columns:
+          raise ValueError(
+              f'{error_prefix} ',
+              f'Same field name "{col}" was specified more than once.')
+        output_fields.append(f'{input}.{col} AS {col}')
+        named_columns.add(col)
+    elif isinstance(cols, dict):
+      for k, v in cols.items():
+        if k in named_columns:
+          raise ValueError(
+              f'{error_prefix} ',
+              f'Same field name "{k}" was specified more than once.')
+        output_fields.append(f'{input}.{v} AS {k}')
+        named_columns.add(k)
+    else:
+      raise ValueError(error_prefix)
+  for table in tables:
+    if table not in fields.keys():
+      output_fields.append(f'{table}.*')
+  return output_fields
+
+
+def _is_connected(edge_list):
+  graph = {}
+  for edge in edge_list:
+    u, v = edge
+    if u not in graph:
+      graph[u] = []
+    if v not in graph:
+      graph[v] = []
+    graph[u].append(v)
+    graph[v].append(u)
```

Comment:
This will only work for equalities of length 2 (due to `u, v = edge`), so the logic will have to take into account input edge lists >2
```suggestion
  graph = {}
  for edge_set in edge_list:
    for u in edge_set:
      if u not in graph:
        graph[u] = set()
      for v in edge_set:
        if u != v:
          graph[u].add(v)
```